// laravel-modal-js/src/Modal.js
export default  {
    modal : null,
    content : null,
    back : null,

    config : {
        delay : 300
    },

    /**
     * Init a modal.
     */
    init: function() {
        this.modal = $('.modal');
        this.content = $('.modal .content');
        this.back = $('.modal .back');

        // Open modal.
        $('a.__js-modal__open').on('click', function(e) {
            e.preventDefault();

            window.Modal.display($(this).attr('href'));
        });

        // Close modal.
        $('.__js-modal__close').on('click', function() {
            window.Modal.close();
        });

        $(document).keyup(function(e) {
            if (e.keyCode == 27) {
                window.Modal.close();
            }
        });
    },

    /**
     * Request a response.
     *
     * @param string href
     */
    display: function(href) {
        $.ajax({
            method : 'POST',
            url : href,
            success : function(response) {
                window.Modal.show(response);
            }
        });
    },

    /**
     * Show modal.
     *
     * @param string html
     */
    show: function(html) {
        $('html, body').animate({
            scrollTop: '0px'
        }, this.config.delay + 200);

        this.content.html(html);
        this.modal.fadeIn(this.config.delay);
    },

    /**
     * Close modal.
     */
    close: function() {
        this.content.html('');
        this.modal.fadeOut(this.config.delay);
    }
}
